utilities/AFQ_mrtrixInit.m

```diff
@@ -80,101 +125,145 @@
-      files.response, files.b, true,[],8,0) % That last 'true' means a figure of the 
-                                  % response function will be displayed
+  AFQ_mrtrix_response(files.brainmask, ...
+                      files.fa, ...
+                      files.sf, ...
+                      files.dwi,...
+                      files.response, ...
+                      files.b, ...
+                      [], ... %this is the threshold string, it was missing!
+                      false, ... % this is show figure
+                      [], ... %bckground
+                      8, ... %lmax
+                      false, ... %verbose
+                      mrtrixVersion) 
 end
 
 % Create a white-matter mask, tracktography will act only in here.
-if (~computed.('wm'))
+if (~computed.('wmMask'))
   wmMaskFile = fullfile(session, dt_info.files.wmMask);
-  mrtrix_mrconvert(wmMaskFile, files.wm,[],0)
+  AFQ_mrtrix_mrconvert(wmMaskFile, ...
+                       files.wmMask, ...
+                       [], ...
+                       0, ...
+                       mrtrixVersion)
 end
 
-% Compute the CSD estimates: 
-if (~computed.('csd'))  
-  disp('The following step takes a while (a few hours)');                                  
-  mrtrix_csdeconv(files.dwi, files.response, lmax, files.csd, files.b, files.brainmask,0)
-end
-
-
 
-%%%%%%%%%%%%%%%%%%%%%%%%%%
-% mrtrix_check_processes %
-%%%%%%%%%%%%%%%%%%%%%%%%%%
-function computed = mrtrix_check_processes(files)
-%
-% Check which mrtrix proceses were computed. Returns an array of 0's and
-% 1's indicating which process was computed (1) and which one needs to be
-% computed (0)
-%
 
-fields = fieldnames(files);
-for ii = 1:length(fields)
-  if exist(files.(fields{ii}),'file') == 2
-    computed.(fields{ii}) = 1;
-  else
-    computed.(fields{ii}) = 0;
-  end
+if ~multishell
+    % Compute the CSD estimates: 
+    if (~computed.('csd'))  
+      disp('The following step takes a while (a few hours)');                                  
+      AFQ_mrtrix_csdeconv(files.dwi, ...
+                          files.response, ...
+                          lmax, ...
+                          files.csd, ... %out
+                          files.b, ... %grad
+                          files.brainmask,... %mask
+                          false,... % Verbose
+                          mrtrixVersion)
+    end
+
+else
+    % Create the 5tt file from the same ac-pc-ed T1 nii we used in the other steps: 
+    if (~computed.('tt5')) && (mrtrixVersion > 2)
+
+        tool = 'freesurfer'; % Options are 'fsl' or 'freesurfer'
+        
+        if strcmp(tool, 'fsl')
+            inputFile = fullfile(session, dt_info.files.t1);
+            if ~(exist(inputFile, 'file') == 2)
+                error(['Cannot find T1, please copy it to ' session]);
+            end              
+        else
+            inputFile = fullfile(session,'aparc+aseg.mgz');
```

Comment:
Yes, I will modify it to accept mgz or nii or nii.gz.
I can add 'tool' as an afq parameter in AFQ_Create and specify that you
will need apart+aseg (or any 'aseg') in order to make it work.
I will make 'fsl' as the default, as it doesn't require extra files (but
for me the results are far worse...)
Regarding the ac-pc question: I understand that you are inputing a T1 that
hasn't been ac-pc-ed so all your DWI pipeline is going to be in freesurfer
space, is that right?

On Thu, Jul 14, 2016 at 2:13 AM dstrodtman <notifications@github.com> wrote:

> In utilities/AFQ_mrtrixInit.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r70730587>:
>
> > +                          mrtrixVersion)
> > +    end
> > +
> > +else
> > +    % Create the 5tt file from the same ac-pc-ed T1 nii we used in the other steps:
> > +    if (~computed.('tt5')) && (mrtrixVersion > 2)
> > +
> > +        tool = 'freesurfer'; % Options are 'fsl' or 'freesurfer'
> > +
> > +        if strcmp(tool, 'fsl')
> > +            inputFile = fullfile(session, dt_info.files.t1);
> > +            if ~(exist(inputFile, 'file') == 2)
> > +                error(['Cannot find T1, please copy it to ' session]);
> > +            end
> > +        else
> > +            inputFile = fullfile(session,'aparc+aseg.mgz');
>
> I'm starting with preprocessed Human Connectome Project data, and have
> aparc+aseg.nii.gz instead in the location specified. Should this file work
> in place of the file you specified, if the code were modified to be more
> robust? I'm looking through mrTrix3 documentation now.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/04eb7b672913be5a6d822faf4964d7e59274eb25#r70730587>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEWQVzGuEdWwLpDgf-fJu5P2tr5UUUrpks5qVX8hgaJpZM4JGbEU>
> .
>